src/libraries/System.ComponentModel.TypeConverter/src/System/ComponentModel/ReflectTypeDescriptionProvider.cs

```diff
@@ -1338,5 +1377,54 @@ private static object SearchIntrinsicTable(Hashtable table, Type callingType)
 
             return hashEntry;
         }
+
+        /// <summary>
+        /// Searches the intrinsic converter dictionary for a match with the object type.
+        /// The strongly-typed dictionary maps object types to converter data objects which lazily
+        /// creates (and caches for re-use, where applicable) converter instances.
+        /// </summary>
+        private static TypeConverter GetIntrinsicTypeConverter(Type callingType)
+        {
+            TypeConverter converter;
+
+            // We take a lock on this dictionary. Nothing in this code calls out to
+            // other methods that lock, so it should be fairly safe to grab this lock.
+            lock (IntrinsicTypeConverters)
+            {
+                if (!IntrinsicTypeConverters.TryGetValue(callingType, out IntrinsicTypeConverterData converterData))
+                {
+                    if (callingType.IsEnum)
+                    {
+                        converterData = IntrinsicTypeConverters[typeof(Enum)];
+                    }
+                    else if (callingType.IsArray)
+                    {
+                        converterData = IntrinsicTypeConverters[typeof(Array)];
+                    }
+                    else if (Nullable.GetUnderlyingType(callingType) != null)
+                    {
+                        converterData = IntrinsicTypeConverters[s_intrinsicNullableKey];
+                    }
+                    else if (typeof(ICollection).IsAssignableFrom(callingType))
+                    {
+                        converterData = IntrinsicTypeConverters[s_intrinsicNullableKey];
```

Comment:
Is using `s_intrinsicNullableKey` here a mistake when the callingType is an `ICollection`? Shouldn't you be using `typeof(ICollection)`?

If a test didn't break for this, we should add one.

Reply:
Yes, thanks for catching this. Adding a test.